fink.info.in

```diff
@@ -15,46 +19,72 @@ Source-MD5: @MD5@
 NoSetPATH: true
 BuildAsNobody: false
 CompileScript: <<
- ./pre-build-test.sh
- make test PREFIX=%p ARCHITECTURE=%m DISTRIBUTION=@DISTRIBUTION@
- ./setup.sh %p %m
+./pre-build-test.sh
+make test PREFIX=%p ARCHITECTURE=%m DISTRIBUTION=@DISTRIBUTION@
+./setup.sh %p %m
 <<
 InstallScript: <<
- ./install.sh %i %m
- install -c -p -m 644 fink.shlibs "%d/DEBIAN/shlibs"
+./install.sh %i %m
+install -c -p -m 644 fink.shlibs "%d/DEBIAN/shlibs"
+<<
+Triggers: <<
```

Comment:
This will produce an invalid .info file after installation now, raising a complaint on
> Error: use of Triggers requires declaring a BuildDepends, Depends, or (rarely) Pre-Depends on:
	fink (>= 0.45.99.git)

but adding that here to `BuildDepends` already breaks in the install phase due to a circular dependency.
Any thoughts how to best deal with this?

Reply:
Just need to ignore the error for the fink package or any package with -bootstrap in it.  Since this is only a bug if you rebuild fink, I'll get around to it once I adopt the require versions into the validator.

Reply:
What does the `Interest-noawait` trigger do? I couldn't understand the deb-triggers.5 man page at all. Does the `Triggers:` field in fink.info.in affect the `"$1" = "triggered"` conditional in `PostInstScript`?
Based on what I think I know what this bit is trying to do, the trigger listens to %p/lib and %p/opt and if a change happens, then PostInstScript cleans any present .la files?

Reply:
yes it's cause the maintainer scripts with a param on triggers.  The different modes are when it's called in fink it won't matter much since dpkg is called directly nothing with only one deb or family of debs.  But with apt-get it'll matter if it runs right away or at the end of all installs.

The thing with triggers is if there is an interest in a dir then even if it's not installing that package but a different package that maintainer script will run.  So things like install-info doesn't need to be in the maintainer script for a package anymore, if the package has a .info file the trigger from fink will run automatically.

Reply:
this is the last issue, we were to move the triggers to dpkg-files IIRC? I just want to make sure I remember this correctly before I make the change.

Reply:
NVM this was moved to dpkg already